src/web/extractors.rs

```diff
@@ -425,46 +440,71 @@ pub struct BsoParam {
 }
 
 impl BsoParam {
-    pub fn bsoparam_from_path(uri: &Uri) -> Result<Self, Error> {
+    pub fn bsoparam_from_path(uri: &Uri, tags: &Tags) -> Result<Self, Error> {
         // TODO: replace with proper path parser
         // path: "/1.5/{uid}/storage/{collection}/{bso}"
         let elements: Vec<&str> = uri.path().split('/').collect();
         let elem = elements.get(3);
         if elem.is_none() || elem != Some(&"storage") || elements.len() != 6 {
-            warn!("⚠️ Unexpected BSO URI: {:?}", uri.path());
+            warn!("⚠️ Unexpected BSO URI: {:?}", uri.path();
+            "ua.os.family" => tags.get("ua.os.family"),
+            "ua.browser.family" => tags.get("ua.browser.family"),
+            "ua.name" => tags.get("ua.name"),
+            "ua.os.ver" => tags.get("ua.os.ver"),
+            "ua.browser.ver" => tags.get("ua.browser.ver"));
             return Err(ValidationErrorKind::FromDetails(
                 "Invalid BSO".to_owned(),
                 RequestErrorLocation::Path,
                 Some("bso".to_owned()),
+                Some(tags.clone()),
             ))?;
         }
         if let Some(v) = elements.get(5) {
-            let sv = String::from_str(v).map_err(|e| {
-                warn!("⚠️ BsoParam Error element:{:?} error:{:?}", v, e);
+            let sv = String::from_str(v).map_err(|_| {
+                // TODO: Why is the param for this block causing warn! to say it's unreachable?
```

Comment:
When does warn! say it's unreachable (compile time or what?)

Reply:
so, after a bit of back and forth, this appears to have been an anomaly. I've not reproduced it. I removed the comment.